File: week11/week11-2.py
BOUND=300000
table=[True]*BOUND
ans=0
for i in range(2,BOUND):
  if table[i]==True:
    #print(i,end=" ")
    ans+=1
    for j in range(i*i,BOUND,i):
      table[j]=False
print("300K的質數有：",ans,"個")

File: week11/week11-2.cpp
#include <stdio.h>
int main()
{
    int BOUND=20000,ans=0;
    int table[20000]={};
    for(int i=2;i<BOUND;i++){
        if(table[i]==0){
            ans++;
            for(int j=i*i;j<BOUND;j+=i){
                table[j]=-1;
            }
        }
    }
    printf("20K的質數： %d 個\n",ans);
}
